pkg/store/consul/transaction/transaction.go

```diff
@@ -35,10 +35,20 @@ type tx struct {
 	committedMu sync.Mutex
 }
 
+// New() returns a new context derived from the one passed as an argument and
+// its cancel function with the context containing metadata used to build a
+// consul transaction. If the passed context already has consul operations
+// defined, the new context will inherit those operations defined but in a
+// separate transaction.
 func New(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
basically it would be like (shortcuts taken for brevity)
```
lockCtx := transaction.New(context.Background())
unlockCtx := transaction.New(context.Background())
checkLockedCtx := transaction.New(context.Background())

buildTxns(lockCtx, unlockCtx, checkLockedCtx)

// acquire the lock
transaction.Commit(lockCtx)

// release lock when work is done
defer transaction.Commit(unlockCtx)

// branch some transactions off the "check locked one"
checkLockedCtx1 := transaction.New(checkLockedTxn)
doSomeWork(checkLockedCtx1)

checkLockedCtx2 := transaction.New(checkLockedTxn)
doSomeWork(checkLockedCtx2)
```

If `Branch()` seems more clear i'm all for that. `Branch()` can fail if there is no transaction in the provided context, and `New()` can fail if there is one?